### PR TITLE
Feature/pre post

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- adding new filter for overriding integration ID
- adding local storage save for field values
- adding prefill for field from url
- bugfixes

### CODE REVIEW

File: testFilters/testFilters.php [hunk 1]
```diff
@@ -808,6 +809,31 @@ public function getWorkableIntegrationPrePostParams(array $params): array
...
+	/**
+	 *
+	 * Change integration ID fields data before we send it to the external integration.
+	 *
+	 * @param string $itemId Item id.
+	 * @param array<mixed> $params Params to alterate.
+	 * @param string $formId Form id.
+	 *
+	 * @return string
+	 */
```
Comment (iobrado) [inline testFilters/testFilters.php:828]:
```suggestion
	/**
	 * Change integration ID fields data before we send it to the external integration.
	 *
	 * @param string $itemId Item id.
	 * @param array<mixed> $params Params to alterate.
	 * @param string $formId Form id.
	 *
	 * @return string
	 */
```

Double docblock comment

**Review by iobrado — CHANGES_REQUESTED** [review]
One minor change

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

I have just one question / suggestion

File: testFilters/testFilters.php (hunk 1 above)
Comment (iruzevic) [reply testFilters/testFilters.php:828]: tnx

File: src/Blocks/components/form/assets/form.js [hunk 2]
```diff
@@ -985,6 +993,7 @@ export class Form {
 			const state = this.state;
 			const utils = this.utils;
 			const conditionalTags = this.conditionalTags;
+			const enrichment = this.enrichment;
 
 			flatpickr.default(input, {
 				enableTime: this.state.getStateElementTypeInternal(name, formId) === 'datetime',
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Blocks/components/form/assets/form.js:999]: yes I have updated it tnx

**Review by kancijan — APPROVED** [review]

File: src/Blocks/components/form/assets/utilities.js [hunk 3]
```diff
@@ -757,30 +767,274 @@ export class Utils {
...
 		const input = this.state.getStateElementInput(name, formId);
-		const choices = this.state.getStateElementCustom(name, formId);
 
-		if (typeof value === 'object') {
-			[...value].forEach((item) => {
-				choices?.removeActiveItemsByValue(item);
-			});
-		} else {
-			choices?.setChoiceByValue(value);
+		console.log(input, custom, value);
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/utilities.js:969]: 👀

File: src/Blocks/components/form/assets/utilities.js [hunk 4]
```diff
@@ -757,30 +767,274 @@ export class Utils {
...
+	 * Set field value - Checkboxes.
+	 * 
+	 * @param {string} formId Form Id.
+	 * @param {string} name Field name.
+	 * @param {object} value Field value.
+	 * 
+	 * @returns {void}
+	 */
+	setCheckboxValue(formId, name, value) {
+		console.log(formId, name, value);
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/utilities.js:988]: 👀

File: src/Blocks/components/form/assets/utilities.js [hunk 5]
```diff
@@ -757,30 +767,274 @@ export class Utils {
...
+
+		Object.entries(value).forEach(([innerName, innerValue]) => {
+			const innerInput = this.state.getStateElementItemsInput(name, innerName, formId);
+			// console.log(innerInput, innerValue);
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/form/assets/utilities.js:992]: 👀

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]

**Pull request merged**